zypp/PurgeKernels.cc

```diff
@@ -255,14 +255,19 @@ namespace zypp {
     const unsigned tokenGrp = 1; //index of the group matching the token
     const unsigned modifierGrp = 2; //index of the group matching the offset modifier
 
+    if (_keepSpec.empty()) {
+	_keepSpec = "latest,latest-1,running";
+    }
 
     MIL << "Parsing keep spec: " << _keepSpec << std::endl;
 
     std::vector<std::string> words;
     str::split( _keepSpec, std::back_inserter(words), ",", str::TRIM );
     if ( words.empty() ) {
-      WAR << "Invalid keep spec: " << _keepSpec << " using default latest,running." << std::endl;
-      return;
+      WAR << "Invalid keep spec: " << _keepSpec << " using default latest,latest-1,running." << std::endl;
```

Comment:
Should we really just fall back to the default keep spec if the admin speficied a incorrect one? That might result in purging kernels and packages that should not be purged. IMO its better to error out in that case.

Reply:
maybe but that would be separate change. The previous code just skipped the parsing in error case so used the global defaults of those variables. Ie equivalent to "latest,running". That's even less than the config file default of "latest,latest-1,running". So this change is for consistency.